tests/unit/oidc/models/test_github.py

```diff
@@ -656,6 +656,26 @@ def test_github_publisher_duplicates_cant_be_created(self, db_request):
             db_request.db.add(publisher2)
             db_request.db.commit()
 
+    @pytest.mark.parametrize(
+        ("url", "expected"),
+        [
+            ("https://repository_owner.github.io/repository_name/", True),
+            ("https://repository_owner.github.io/repository_name", True),
+            ("https://repository_owner.github.io/repository_name/subpage", True),
+            ("https://repository_owner.github.io/repository_name/../malicious", False),
+            ("https://repository_owner.github.io/", False),
```

Comment:
Might want a testcase for an unrelated subpage as well, e.g.:

```suggestion
            ("https://repository_owner.github.io/unrelated_name/", False),
```